dz13/src/app.js

```diff
@@ -0,0 +1,126 @@
+function entriesArray(array) {
+  return array.flat(Infinity)
+}
+
+// function setKeysAndObjectInfo(user) {
+//   const entriesUser = Object.entries(user)
+//   const keyUser = []
+//   for (const key of entriesUser) {
+//     key.pop()
+//     keyUser.push(key)
+//   }
+//   console.log(keyUser.flat())
+// }
+function setKeysAndObjectInfo(user, type) {
+  if (type === 'key') {
+    const newUser = Object.keys(user)
+    return newUser
+  }
+  const newUser = Object.values(user)
+  return newUser
+}
```

Comment:
```suggestion
function getObjectKeysOrValues(object, type) {
  return type === 'key' ? Object.keys(object) : Object.values(object)
}
```

- не нужно создавать лишние переменные, можно сразу возвращать результат выполнения
- функции, которые начинаются с `set`, обычно подразумевают, что что-то задаётся. которые возвращают обычно начинаются с `get`
- т.к. из названия функции возвращаются ключи или значения у любого объекта, то и параметр функции логично назвать любой объект `object` вместо конкретного `user`